### PR TITLE
Fix `ThrowsUncheckedException` Error Prone violations in tests

### PR DESCRIPTION
**Pull request by basil**
Fix violations of the [`ThrowsUncheckedException`](https://errorprone.info/bugpattern/ThrowsUncheckedException) Error Prone check in tests. Unchecked exceptions do not need to be declared in the method signature. [Effective Java 3rd Edition §74](https://books.google.com/books?id=BIpDDwAAQBAJ) says:

> Use the Javadoc `@throws` tag to document each exception that a method can throw, but do _not_ use the `throws` keyword on unchecked exceptions.

We don't generate Javadoc for tests, so this suggestion can be implemented by simply removing any unchecked exceptions from `throws` statements.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]
Should be fine, although IMO the justification here seems a bit weak. Assuming the content didn't change substantially between second (which is in front of me, _Item 62: Document all exceptions thrown by each method_) and third edition, it emphasizes the need to document all exceptions thrown in Javadoc, basically making the `throws` redundant and therefore unnecessary.

**Comment by basil** [comment]
> Should be fine, although IMO the justification here seems a bit weak. Assuming the content didn't change substantially between second (which is in front of me, _Item 62: Document all exceptions thrown by each method_) and third edition, it emphasizes the need to document all exceptions thrown in Javadoc, basically making the `throws` redundant and therefore unnecessary.

Good point. I rolled back the changes to overridden methods, since the `throws` would presumably be documented there in the ideal scenario. What remains are things like `throws AssertionError` on private methods, which definitely seems unnecessary to me.

**Review by res0nance — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
